service/src/main/kotlin/fi/espoo/evaka/nekku/NekkuQueries.kt

```diff
@@ -0,0 +1,87 @@
+// SPDX-FileCopyrightText: 2017-2025 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+package fi.espoo.evaka.nekku
+
+import fi.espoo.evaka.decision.logger
+import fi.espoo.evaka.shared.db.Database
+import org.jdbi.v3.core.mapper.PropagateNull
+
+data class CustomerNumbers(
+    @PropagateNull val number: String,
+    val name: String,
+    val customergroup: String,
+    val unit_size: String,
+)
+
+/** Throws an IllegalStateException if Nekku returns an empty texture list. */
+fun fetchAndUpdateNekkuCustomers(
+    client: NekkuClient,
+    db: Database.Connection,
+    warner: (s: String) -> Unit = loggerWarner,
+) {
+    val customersFromNekku =
+        client.getCustomers().map { it ->
+            CustomerNumbers(it.number, it.name, it.customergroup, it.unit_size)
+        }
+
+    if (customersFromNekku.isEmpty())
+        error("Refusing to sync empty Nekku customer list into database")
+    db.transaction { tx ->
+        val nulledCustomersCount =
+            tx.resetNekkuCustomerNumbersNotContainedWithin(customersFromNekku)
+        if (nulledCustomersCount != 0)
+            warner(
```

Comment:
Erillistä `warner` parametria ei tarvita. Voit käyttää suoraan `logger.warn(...`. Jamix intagraation synkronaatio koodissa tuo erillinen `warner` oli tarpeen, jotta voitiin testeissä tarkistaa, että logille tulee tarpeeksi suurella levelillä viesti jos käytössäolevia diettejä poistuu. Myöhemmin sinne lisättiin myös varoitussähköpostin lähetys jos näin tapahtuu.